extensions/barrage/src/main/java/io/deephaven/extensions/barrage/table/BarrageTable.java

```diff
@@ -271,6 +259,13 @@ public void handleBarrageError(Throwable t) {
         enqueueError(t);
     }
 
+    private synchronized void tryToDeliverErrorToCallback(final Throwable err) {
```

Comment:
I think we want to synchronize `updateServerViewport`, too. Or at least the part that deals with the callback.